extension/js/common/core/crypto/pgp/openpgp-key.ts

```diff
@@ -468,6 +484,50 @@ export class OpenPGPKey {
     return secondTry ? secondTry : null; // tslint:disable-line:no-null-keyword
   }
 
+  private static getSigningAndEncryptionKeys = async (key: OpenPGP.key.Key, exp: number | Date | null, expired: () => boolean) => {
+    const getEncryptionKey = (keyid?: OpenPGP.Keyid | null, date?: Date, userId?: OpenPGP.UserId | null) =>
+      key.getEncryptionKey(keyid, date, userId);
+    const encryptionKey = await Catch.undefinedOnException(getEncryptionKey());
+    const encryptionKeyIgnoringExpiration = encryptionKey ? encryptionKey : await OpenPGPKey.getKeyIgnoringExpiration(getEncryptionKey, exp, expired);
+    const getSigningKey = (keyid?: OpenPGP.Keyid | null, date?: Date, userId?: OpenPGP.UserId | null) =>
+      key.getSigningKey(keyid, date, userId);
+    const signingKey = await Catch.undefinedOnException(getSigningKey());
+    const signingKeyIgnoringExpiration = signingKey ? signingKey : await OpenPGPKey.getKeyIgnoringExpiration(getSigningKey, exp, expired);
+    return { encryptionKey, encryptionKeyIgnoringExpiration, signingKey, signingKeyIgnoringExpiration };
+  }
+
+  private static removeInsecureKeyPackets = (opgpKey: OpenPGP.key.Key): { primaryKeySecure: boolean, secure: OpenPGP.key.Key } => {
+    let primaryKeySecure = true;
+    const packets = opgpKey.toPacketlist();
+    const newPacketList = new opgp.packet.List<OpenPGP.packet.BasePacket>();
+    for (const packet of packets) {
+      if (OpenPGPKey.isBaseKeyPacket(packet)) {
+        const { algorithm, bits } = packet.getAlgorithmInfo();
+        if (!OpenPGPKey.minimumBitsByAlgo) {
+          OpenPGPKey.minimumBitsByAlgo = {
+            [opgp.enums.read(opgp.enums.publicKey, opgp.enums.publicKey.rsa_encrypt)]: 2048,
+            [opgp.enums.read(opgp.enums.publicKey, opgp.enums.publicKey.rsa_encrypt_sign)]: 2048,
+            [opgp.enums.read(opgp.enums.publicKey, opgp.enums.publicKey.rsa_sign)]: 2048,
+          };
+        }
+        const minimumBits = OpenPGPKey.minimumBitsByAlgo[algorithm];
+        if (minimumBits && bits < minimumBits) {
+          if (packet === opgpKey.primaryKey) {
+            // the primary key packet should remain, otherwise the key can't be parsed
+            primaryKeySecure = false;
+          } else {
+            continue; // discard this packet as insecure
+          }
+        }
+      }
+      newPacketList.push(packet);
+    }
+    if (packets.length !== newPacketList.length) {
+      return { primaryKeySecure, secure: new opgp.key.Key(newPacketList) };
+    }
+    return { primaryKeySecure, secure: opgpKey };
```

Comment:
Let's rename the method to `removeWeakKeyPackets` and return values to `{isPrimaryKeyStrong, keyWithoutWeakPackets}`

Please add method comment/doc to link to the github issue 2715 for further reading / understanding